quilt/tools/store.py

```diff
@@ -141,14 +176,34 @@ def ls_packages(self):
         """
         List packages in this store.
         """
-        packages = [
-            (user, pkg[:-len(self.PACKAGE_FILE_EXT)])
-            for user in os.listdir(self._path)
-            if os.path.isdir(os.path.join(self._path, user))
-            for pkg in os.listdir(os.path.join(self._path, user))
-            if pkg.endswith(self.PACKAGE_FILE_EXT)]
+        packages = []
+        pkgdir = os.path.join(self._path, self.PKG_DIR)
+        for user in os.listdir(pkgdir):
+            for pkg in os.listdir(os.path.join(pkgdir, user)):
+                pkgpath = os.path.join(pkgdir, user, pkg)           
+                pkgmap = {h : None for h in os.listdir(os.path.join(pkgpath, Package.CONTENTS_DIR))}
+                for tag in os.listdir(os.path.join(pkgpath, Package.TAGS_DIR)):
+                    with open(os.path.join(pkgpath, Package.TAGS_DIR, tag), 'r') as tagfile:
+                        pkghash = tagfile.read()
+                        pkgmap[pkghash] = tag
```

Comment:
Can there be multiple tags for a given hash?

Reply:
Yes. You can have many tags for the same hash.

Reply:
But they'll overwrite each other here, won't they?

Also, how would I create new tags? `quilt tag add ...` should probably do that - but it's only creating server-side tags right now.

Reply:
I've left updating the commands (e.g., tag add) to another PR.